src/components/search/search.js

```diff
@@ -19,9 +20,28 @@ class SearchComponent extends React.Component {
     this.setState({ postcode: event.target.value });
   }
 
-  doPostcodeSearch = props => {
-    this.setState({ doSearch: true });
-  };
+  doPostcodeSearch() {
+    console.log(this.state.postcode);
+    if (this.state.postcode) {
+      let url =
+        "http://localhost:8000/api/venue/?postcode=" + this.state.postcode;
+
+      fetch(url)
+        .then(response => {
+          return response.json();
+        })
+        .then(data => {
+          if (data.detail) {
+            console.log(data.detail);
+            this.setState({ error: data.detail });
```

Comment:
It's not obvious to me why we need to set error with the data.detail, could you elaborate what this code is doing?

Reply:
The server returns data in this format. I have added a comment into my next commit